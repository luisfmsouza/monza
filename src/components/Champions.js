import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { getRacesBySeason } from "./utils";
import RaceWinners from "./RaceWinners";

const ChampionRow = styled.div`
  padding: 30px;
  margin: 30px;
  background-color: ${({ theme }) => theme.colors.grey30};
  display: grid;
  grid-template-columns: auto 50px;
  grid-template-areas:
    "year button"
    "name button";
`;

const Year = styled.span`
  grid-area: year;
`;
const Name = styled.span`
  font-size: 40px;
  font-weight: 600;
  grid-area: name;
`;

const Button = styled.button`
  align-self: center;
  background-color: ${({ theme }) => theme.colors.blue10};
  border: none;
  cursor: pointer;
  grid-area: button;
  height: 50px;
`;

const ArrowDown = () => (
  <svg width="32" height="32">
    <defs>
      <path
        d="M15 19.657V8a1 1 0 012 0v11.657l2.95-2.95a1 1 0 111.414 1.414l-4.243 4.243a.998.998 0 01-.3.206.999.999 0 01-1.642 0 .998.998 0 01-.3-.206l-4.243-4.243a1 1 0 111.414-1.414l2.95 2.95z"
        id="a"
      />
    </defs>
    <use fill="#fff" fillRule="nonzero" xlinkHref="#a" />
  </svg>
);

const Champions = ({ standingsList, raceList, requestYearResults }) => (
  <div>
    {standingsList.map(season => (
      <ChampionRow key={season.season}>
        <Year>{season.season}</Year>
        <Name>{`${season.driver.givenName} ${season.driver.familyName}`}</Name>

        <RaceWinners
          raceListBySeason={getRacesBySeason(raceList, season.season)}
          season={season.season}
          champion={season.driver.code}
        />

        <Button onClick={() => requestYearResults(season.season)}>
          <ArrowDown />
        </Button>
      </ChampionRow>
    ))}
  </div>
);

Champions.propTypes = {
  standingsList: PropTypes.arrayOf(
    PropTypes.shape({
      season: PropTypes.string.isRequired,
      driver: PropTypes.shape({
        driverId: PropTypes.string.isRequired,
        permanentNumber: PropTypes.string.isRequired,
        code: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired,
        givenName: PropTypes.string.isRequired,
        familyName: PropTypes.string.isRequired,
        dateOfBirth: PropTypes.string.isRequired,
        nationality: PropTypes.string.isRequired
      })
    })
  ),
  raceList: PropTypes.arrayOf(
    PropTypes.shape({
      season: PropTypes.string.isRequired,
      races: PropTypes.arrayOf(
        PropTypes.shape({
          driverId: PropTypes.string.isRequired,
          code: PropTypes.string.isRequired,
          url: PropTypes.string.isRequired,
          givenName: PropTypes.string.isRequired,
          familyName: PropTypes.string.isRequired,
          dateOfBirth: PropTypes.string.isRequired,
          nationality: PropTypes.string.isRequired,
          circuitName: PropTypes.string,
          date: PropTypes.string
        })
      ).isRequired
    })
  ),
  requestYearResults: PropTypes.func.isRequired
};

export default Champions;
