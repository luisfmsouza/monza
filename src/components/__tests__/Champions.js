import React from "react";
import { shallow } from "enzyme";

import Champions from "../Champions";

describe("Champions", () => {
  it("renders without crashing", () => {
    const props = {
      standingsList: [
        {
          season: "2005",
          driver: {
            driverId: "alonso",
            permanentNumber: "14",
            code: "ALO",
            url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
            givenName: "Fernando",
            familyName: "Alonso",
            dateOfBirth: "1981-07-29",
            nationality: "Spanish"
          }
        },
        {
          season: "2006",
          driver: {
            driverId: "alonso",
            permanentNumber: "14",
            code: "ALO",
            url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
            givenName: "Fernando",
            familyName: "Alonso",
            dateOfBirth: "1981-07-29",
            nationality: "Spanish"
          }
        },
        {
          season: "2007",
          driver: {
            driverId: "raikkonen",
            permanentNumber: "7",
            code: "RAI",
            url: "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
            givenName: "Kimi",
            familyName: "Räikkönen",
            dateOfBirth: "1979-10-17",
            nationality: "Finnish"
          }
        },
        {
          season: "2008",
          driver: {
            driverId: "hamilton",
            permanentNumber: "44",
            code: "HAM",
            url: "http://en.wikipedia.org/wiki/Lewis_Hamilton",
            givenName: "Lewis",
            familyName: "Hamilton",
            dateOfBirth: "1985-01-07",
            nationality: "British"
          }
        },
        {
          season: "2009",
          driver: {
            driverId: "button",
            permanentNumber: "22",
            code: "BUT",
            url: "http://en.wikipedia.org/wiki/Jenson_Button",
            givenName: "Jenson",
            familyName: "Button",
            dateOfBirth: "1980-01-19",
            nationality: "British"
          }
        },
        {
          season: "2010",
          driver: {
            driverId: "vettel",
            permanentNumber: "5",
            code: "VET",
            url: "http://en.wikipedia.org/wiki/Sebastian_Vettel",
            givenName: "Sebastian",
            familyName: "Vettel",
            dateOfBirth: "1987-07-03",
            nationality: "German"
          }
        },
        {
          season: "2011",
          driver: {
            driverId: "vettel",
            permanentNumber: "5",
            code: "VET",
            url: "http://en.wikipedia.org/wiki/Sebastian_Vettel",
            givenName: "Sebastian",
            familyName: "Vettel",
            dateOfBirth: "1987-07-03",
            nationality: "German"
          }
        },
        {
          season: "2012",
          driver: {
            driverId: "vettel",
            permanentNumber: "5",
            code: "VET",
            url: "http://en.wikipedia.org/wiki/Sebastian_Vettel",
            givenName: "Sebastian",
            familyName: "Vettel",
            dateOfBirth: "1987-07-03",
            nationality: "German"
          }
        },
        {
          season: "2013",
          driver: {
            driverId: "vettel",
            permanentNumber: "5",
            code: "VET",
            url: "http://en.wikipedia.org/wiki/Sebastian_Vettel",
            givenName: "Sebastian",
            familyName: "Vettel",
            dateOfBirth: "1987-07-03",
            nationality: "German"
          }
        },
        {
          season: "2014",
          driver: {
            driverId: "hamilton",
            permanentNumber: "44",
            code: "HAM",
            url: "http://en.wikipedia.org/wiki/Lewis_Hamilton",
            givenName: "Lewis",
            familyName: "Hamilton",
            dateOfBirth: "1985-01-07",
            nationality: "British"
          }
        },
        {
          season: "2015",
          driver: {
            driverId: "hamilton",
            permanentNumber: "44",
            code: "HAM",
            url: "http://en.wikipedia.org/wiki/Lewis_Hamilton",
            givenName: "Lewis",
            familyName: "Hamilton",
            dateOfBirth: "1985-01-07",
            nationality: "British"
          }
        }
      ],
      raceList: [
        {
          season: "2005",
          races: [
            {
              driverId: "fisichella",
              code: "FIS",
              url: "http://en.wikipedia.org/wiki/Giancarlo_Fisichella",
              givenName: "Giancarlo",
              familyName: "Fisichella",
              dateOfBirth: "1973-01-14",
              nationality: "Italian"
            },
            {
              driverId: "alonso",
              permanentNumber: "14",
              code: "ALO",
              url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
              givenName: "Fernando",
              familyName: "Alonso",
              dateOfBirth: "1981-07-29",
              nationality: "Spanish"
            },
            {
              driverId: "alonso",
              permanentNumber: "14",
              code: "ALO",
              url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
              givenName: "Fernando",
              familyName: "Alonso",
              dateOfBirth: "1981-07-29",
              nationality: "Spanish"
            },
            {
              driverId: "alonso",
              permanentNumber: "14",
              code: "ALO",
              url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
              givenName: "Fernando",
              familyName: "Alonso",
              dateOfBirth: "1981-07-29",
              nationality: "Spanish"
            },
            {
              driverId: "raikkonen",
              permanentNumber: "7",
              code: "RAI",
              url: "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
              givenName: "Kimi",
              familyName: "Räikkönen",
              dateOfBirth: "1979-10-17",
              nationality: "Finnish"
            },
            {
              driverId: "raikkonen",
              permanentNumber: "7",
              code: "RAI",
              url: "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
              givenName: "Kimi",
              familyName: "Räikkönen",
              dateOfBirth: "1979-10-17",
              nationality: "Finnish"
            },
            {
              driverId: "alonso",
              permanentNumber: "14",
              code: "ALO",
              url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
              givenName: "Fernando",
              familyName: "Alonso",
              dateOfBirth: "1981-07-29",
              nationality: "Spanish"
            },
            {
              driverId: "raikkonen",
              permanentNumber: "7",
              code: "RAI",
              url: "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
              givenName: "Kimi",
              familyName: "Räikkönen",
              dateOfBirth: "1979-10-17",
              nationality: "Finnish"
            },
            {
              driverId: "michael_schumacher",
              code: "MSC",
              url: "http://en.wikipedia.org/wiki/Michael_Schumacher",
              givenName: "Michael",
              familyName: "Schumacher",
              dateOfBirth: "1969-01-03",
              nationality: "German"
            },
            {
              driverId: "alonso",
              permanentNumber: "14",
              code: "ALO",
              url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
              givenName: "Fernando",
              familyName: "Alonso",
              dateOfBirth: "1981-07-29",
              nationality: "Spanish"
            },
            {
              driverId: "montoya",
              code: "MON",
              url: "http://en.wikipedia.org/wiki/Juan_Pablo_Montoya",
              givenName: "Juan",
              familyName: "Pablo Montoya",
              dateOfBirth: "1975-09-20",
              nationality: "Colombian"
            },
            {
              driverId: "alonso",
              permanentNumber: "14",
              code: "ALO",
              url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
              givenName: "Fernando",
              familyName: "Alonso",
              dateOfBirth: "1981-07-29",
              nationality: "Spanish"
            },
            {
              driverId: "raikkonen",
              permanentNumber: "7",
              code: "RAI",
              url: "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
              givenName: "Kimi",
              familyName: "Räikkönen",
              dateOfBirth: "1979-10-17",
              nationality: "Finnish"
            },
            {
              driverId: "raikkonen",
              permanentNumber: "7",
              code: "RAI",
              url: "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
              givenName: "Kimi",
              familyName: "Räikkönen",
              dateOfBirth: "1979-10-17",
              nationality: "Finnish"
            },
            {
              driverId: "montoya",
              code: "MON",
              url: "http://en.wikipedia.org/wiki/Juan_Pablo_Montoya",
              givenName: "Juan",
              familyName: "Pablo Montoya",
              dateOfBirth: "1975-09-20",
              nationality: "Colombian"
            },
            {
              driverId: "raikkonen",
              permanentNumber: "7",
              code: "RAI",
              url: "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
              givenName: "Kimi",
              familyName: "Räikkönen",
              dateOfBirth: "1979-10-17",
              nationality: "Finnish"
            },
            {
              driverId: "montoya",
              code: "MON",
              url: "http://en.wikipedia.org/wiki/Juan_Pablo_Montoya",
              givenName: "Juan",
              familyName: "Pablo Montoya",
              dateOfBirth: "1975-09-20",
              nationality: "Colombian"
            },
            {
              driverId: "raikkonen",
              permanentNumber: "7",
              code: "RAI",
              url: "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
              givenName: "Kimi",
              familyName: "Räikkönen",
              dateOfBirth: "1979-10-17",
              nationality: "Finnish"
            },
            {
              driverId: "alonso",
              permanentNumber: "14",
              code: "ALO",
              url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
              givenName: "Fernando",
              familyName: "Alonso",
              dateOfBirth: "1981-07-29",
              nationality: "Spanish"
            }
          ]
        }
      ],
      requestYearResults: jest.fn()
    };
    const wrapper = shallow(<Champions {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
