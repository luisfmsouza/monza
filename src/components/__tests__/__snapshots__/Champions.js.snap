// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Champions renders without crashing 1`] = `
<div>
  <styled.div
    key="2005"
  >
    <styled.span>
      2005
    </styled.span>
    <styled.span>
      Fernando Alonso
    </styled.span>
    <RaceWinners
      champion="ALO"
      raceListBySeason={
        Object {
          "races": Array [
            Object {
              "code": "FIS",
              "dateOfBirth": "1973-01-14",
              "driverId": "fisichella",
              "familyName": "Fisichella",
              "givenName": "Giancarlo",
              "nationality": "Italian",
              "url": "http://en.wikipedia.org/wiki/Giancarlo_Fisichella",
            },
            Object {
              "code": "ALO",
              "dateOfBirth": "1981-07-29",
              "driverId": "alonso",
              "familyName": "Alonso",
              "givenName": "Fernando",
              "nationality": "Spanish",
              "permanentNumber": "14",
              "url": "http://en.wikipedia.org/wiki/Fernando_Alonso",
            },
            Object {
              "code": "ALO",
              "dateOfBirth": "1981-07-29",
              "driverId": "alonso",
              "familyName": "Alonso",
              "givenName": "Fernando",
              "nationality": "Spanish",
              "permanentNumber": "14",
              "url": "http://en.wikipedia.org/wiki/Fernando_Alonso",
            },
            Object {
              "code": "ALO",
              "dateOfBirth": "1981-07-29",
              "driverId": "alonso",
              "familyName": "Alonso",
              "givenName": "Fernando",
              "nationality": "Spanish",
              "permanentNumber": "14",
              "url": "http://en.wikipedia.org/wiki/Fernando_Alonso",
            },
            Object {
              "code": "RAI",
              "dateOfBirth": "1979-10-17",
              "driverId": "raikkonen",
              "familyName": "Räikkönen",
              "givenName": "Kimi",
              "nationality": "Finnish",
              "permanentNumber": "7",
              "url": "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
            },
            Object {
              "code": "RAI",
              "dateOfBirth": "1979-10-17",
              "driverId": "raikkonen",
              "familyName": "Räikkönen",
              "givenName": "Kimi",
              "nationality": "Finnish",
              "permanentNumber": "7",
              "url": "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
            },
            Object {
              "code": "ALO",
              "dateOfBirth": "1981-07-29",
              "driverId": "alonso",
              "familyName": "Alonso",
              "givenName": "Fernando",
              "nationality": "Spanish",
              "permanentNumber": "14",
              "url": "http://en.wikipedia.org/wiki/Fernando_Alonso",
            },
            Object {
              "code": "RAI",
              "dateOfBirth": "1979-10-17",
              "driverId": "raikkonen",
              "familyName": "Räikkönen",
              "givenName": "Kimi",
              "nationality": "Finnish",
              "permanentNumber": "7",
              "url": "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
            },
            Object {
              "code": "MSC",
              "dateOfBirth": "1969-01-03",
              "driverId": "michael_schumacher",
              "familyName": "Schumacher",
              "givenName": "Michael",
              "nationality": "German",
              "url": "http://en.wikipedia.org/wiki/Michael_Schumacher",
            },
            Object {
              "code": "ALO",
              "dateOfBirth": "1981-07-29",
              "driverId": "alonso",
              "familyName": "Alonso",
              "givenName": "Fernando",
              "nationality": "Spanish",
              "permanentNumber": "14",
              "url": "http://en.wikipedia.org/wiki/Fernando_Alonso",
            },
            Object {
              "code": "MON",
              "dateOfBirth": "1975-09-20",
              "driverId": "montoya",
              "familyName": "Pablo Montoya",
              "givenName": "Juan",
              "nationality": "Colombian",
              "url": "http://en.wikipedia.org/wiki/Juan_Pablo_Montoya",
            },
            Object {
              "code": "ALO",
              "dateOfBirth": "1981-07-29",
              "driverId": "alonso",
              "familyName": "Alonso",
              "givenName": "Fernando",
              "nationality": "Spanish",
              "permanentNumber": "14",
              "url": "http://en.wikipedia.org/wiki/Fernando_Alonso",
            },
            Object {
              "code": "RAI",
              "dateOfBirth": "1979-10-17",
              "driverId": "raikkonen",
              "familyName": "Räikkönen",
              "givenName": "Kimi",
              "nationality": "Finnish",
              "permanentNumber": "7",
              "url": "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
            },
            Object {
              "code": "RAI",
              "dateOfBirth": "1979-10-17",
              "driverId": "raikkonen",
              "familyName": "Räikkönen",
              "givenName": "Kimi",
              "nationality": "Finnish",
              "permanentNumber": "7",
              "url": "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
            },
            Object {
              "code": "MON",
              "dateOfBirth": "1975-09-20",
              "driverId": "montoya",
              "familyName": "Pablo Montoya",
              "givenName": "Juan",
              "nationality": "Colombian",
              "url": "http://en.wikipedia.org/wiki/Juan_Pablo_Montoya",
            },
            Object {
              "code": "RAI",
              "dateOfBirth": "1979-10-17",
              "driverId": "raikkonen",
              "familyName": "Räikkönen",
              "givenName": "Kimi",
              "nationality": "Finnish",
              "permanentNumber": "7",
              "url": "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
            },
            Object {
              "code": "MON",
              "dateOfBirth": "1975-09-20",
              "driverId": "montoya",
              "familyName": "Pablo Montoya",
              "givenName": "Juan",
              "nationality": "Colombian",
              "url": "http://en.wikipedia.org/wiki/Juan_Pablo_Montoya",
            },
            Object {
              "code": "RAI",
              "dateOfBirth": "1979-10-17",
              "driverId": "raikkonen",
              "familyName": "Räikkönen",
              "givenName": "Kimi",
              "nationality": "Finnish",
              "permanentNumber": "7",
              "url": "http://en.wikipedia.org/wiki/Kimi_R%C3%A4ikk%C3%B6nen",
            },
            Object {
              "code": "ALO",
              "dateOfBirth": "1981-07-29",
              "driverId": "alonso",
              "familyName": "Alonso",
              "givenName": "Fernando",
              "nationality": "Spanish",
              "permanentNumber": "14",
              "url": "http://en.wikipedia.org/wiki/Fernando_Alonso",
            },
          ],
          "season": "2005",
        }
      }
      season="2005"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2006"
  >
    <styled.span>
      2006
    </styled.span>
    <styled.span>
      Fernando Alonso
    </styled.span>
    <RaceWinners
      champion="ALO"
      raceListBySeason={null}
      season="2006"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2007"
  >
    <styled.span>
      2007
    </styled.span>
    <styled.span>
      Kimi Räikkönen
    </styled.span>
    <RaceWinners
      champion="RAI"
      raceListBySeason={null}
      season="2007"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2008"
  >
    <styled.span>
      2008
    </styled.span>
    <styled.span>
      Lewis Hamilton
    </styled.span>
    <RaceWinners
      champion="HAM"
      raceListBySeason={null}
      season="2008"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2009"
  >
    <styled.span>
      2009
    </styled.span>
    <styled.span>
      Jenson Button
    </styled.span>
    <RaceWinners
      champion="BUT"
      raceListBySeason={null}
      season="2009"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2010"
  >
    <styled.span>
      2010
    </styled.span>
    <styled.span>
      Sebastian Vettel
    </styled.span>
    <RaceWinners
      champion="VET"
      raceListBySeason={null}
      season="2010"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2011"
  >
    <styled.span>
      2011
    </styled.span>
    <styled.span>
      Sebastian Vettel
    </styled.span>
    <RaceWinners
      champion="VET"
      raceListBySeason={null}
      season="2011"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2012"
  >
    <styled.span>
      2012
    </styled.span>
    <styled.span>
      Sebastian Vettel
    </styled.span>
    <RaceWinners
      champion="VET"
      raceListBySeason={null}
      season="2012"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2013"
  >
    <styled.span>
      2013
    </styled.span>
    <styled.span>
      Sebastian Vettel
    </styled.span>
    <RaceWinners
      champion="VET"
      raceListBySeason={null}
      season="2013"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2014"
  >
    <styled.span>
      2014
    </styled.span>
    <styled.span>
      Lewis Hamilton
    </styled.span>
    <RaceWinners
      champion="HAM"
      raceListBySeason={null}
      season="2014"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
  <styled.div
    key="2015"
  >
    <styled.span>
      2015
    </styled.span>
    <styled.span>
      Lewis Hamilton
    </styled.span>
    <RaceWinners
      champion="HAM"
      raceListBySeason={null}
      season="2015"
    />
    <styled.button
      onClick={[Function]}
    >
      <ArrowDown />
    </styled.button>
  </styled.div>
</div>
`;
